api/src/mutations/mint.rs

```diff
@@ -162,8 +161,116 @@ impl Mutation {
+        let producer = ctx.data::<Producer<NftEvents>>()?;
+        let conn = db.get();
+        let solana = ctx.data::<Solana>()?;
+
+        let UserID(id) = user_id;
+        let user_id = id.ok_or_else(|| Error::new("X-USER-ID header not found"))?;
+
+        let (collection_mint_model, drop) = collection_mints::Entity::find()
+            .join(
+                JoinType::InnerJoin,
+                collection_mints::Relation::Collections.def(),
+            )
+            .join(JoinType::InnerJoin, collections::Relation::Drops.def())
+            .select_also(drops::Entity)
+            .filter(collection_mints::Column::Id.eq(input.mint_id))
+            .one(conn)
+            .await?
+            .ok_or_else(|| Error::new("collection mint not found"))?;
+
+        let collection = collections::Entity::find()
+            .filter(collections::Column::Id.eq(collection_mint_model.collection_id.clone()))
+            .one(conn)
+            .await?
+            .ok_or_else(|| Error::new("collection not found"))?;
+
+        let drop_model = drop.ok_or_else(|| Error::new("drop not found"))?;
+
+        let recipient = collection_mint_model.owner.clone();
+        let edition = collection_mint_model.edition.clone();
+
+        // Fetch the project wallet address which will sign the transaction by hub-treasuries
+        let wallet = project_wallets::Entity::find()
+            .filter(
+                project_wallets::Column::ProjectId
+                    .eq(drop_model.project_id)
+                    .and(project_wallets::Column::Blockchain.eq(collection.blockchain)),
+            )
+            .one(conn)
+            .await?;
+
+        let owner_address = wallet
+            .ok_or_else(|| {
+                Error::new(format!(
+                    "no project wallet found for {} blockchain",
+                    collection.blockchain
+                ))
+            })?
+            .wallet_address;
+
+        let (
+            _,
+            TransactionResponse {
+                serialized_message,
+                signed_message_signatures,
+            },
+        ) = match collection.blockchain {
+            BlockchainEnum::Solana => {
+                solana
+                    .edition()
+                    .mint(CreateEditionRequest {
+                        collection: collection.id,
+                        recipient,
+                        owner_address,
+                        edition: edition.try_into()?,
+                    })
+                    .await?
+            },
+            BlockchainEnum::Polygon | BlockchainEnum::Ethereum => {
+                return Err(Error::new("blockchain not supported as this time"));
+            },
+        };
+
+        let proto_blockchain_enum: proto::Blockchain = collection.blockchain.into();
+
+        MetadataJson::fetch(collection.id, db)
+            .await?
+            .save(collection_mint_model.id, db)
+            .await?;
```

Comment:
I don't believe we need to re-save the metadata json. When we retry it should be from the state when it was initiated.